File: base.service/src/main/java/alvin/base/service/ApplicationComponent.java
package alvin.adv.service;

import javax.inject.Singleton;

import alvin.adv.service.bind.BindModule;
import alvin.adv.service.foreground.ForegroundModule;
import alvin.adv.service.intent.IntentModule;
import alvin.adv.service.lifecycle.LifecycleModule;
import alvin.adv.service.main.MainModule;
import alvin.adv.service.remote.RemoteModule;
import alvin.adv.service.working.WorkingModule;
import dagger.Component;
import dagger.android.AndroidInjector;
import dagger.android.support.AndroidSupportInjectionModule;

@Singleton
@Component(modules = {
        AndroidSupportInjectionModule.class,
        ApplicationModule.class,
        MainModule.class,
        LifecycleModule.class,
        WorkingModule.class,
        BindModule.class,
        IntentModule.class,
        RemoteModule.class,
        ForegroundModule.class
})
public interface ApplicationComponent extends AndroidInjector<Application> {

    @Component.Builder
    abstract class Builder extends AndroidInjector.Builder<Application> { }
}

File: adv.permission/src/main/java/alvin/adv/permission/ApplicationComponent.kt
package alvin.adv.permission

import alvin.adv.permission.main.MainModule
import alvin.adv.permission.remoteservice.RemoteServiceModule
import alvin.adv.permission.storage.StorageModule
import dagger.Component
import dagger.android.AndroidInjector
import dagger.android.support.AndroidSupportInjectionModule
import javax.inject.Singleton

@Singleton
@Component(modules = [
    AndroidSupportInjectionModule::class,
    ApplicationModule::class,
    MainModule::class,
    StorageModule::class,
    RemoteServiceModule::class
])
interface ApplicationComponent : AndroidInjector<Application> {

    @Component.Builder
    abstract class Builder : AndroidInjector.Builder<Application>()
}
